core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java

```diff
@@ -139,7 +145,7 @@ public void listStatus(ListStatusPRequest request,
       responseObserver.onCompleted();
     } catch (Exception e) {
       LOG.error(String.format("Failed to list status of %s: ", request.getPath()), e);
-      responseObserver.onError(e);
+      responseObserver.onError(new io.grpc.StatusException(Status.UNKNOWN));
```

Comment:
```
  /**
   * Receives a terminating error from the stream.
   *
   * <p>May only be called once and if called it must be the last method called. In particular if an
   * exception is thrown by an implementation of {@code onError} no further calls to any method are
   * allowed.
   *
   * <p>{@code t} should be a {@link io.grpc.StatusException} or {@link
   * io.grpc.StatusRuntimeException}, but other {@code Throwable} types are possible. Callers should
   * generally convert from a {@link io.grpc.Status} via {@link io.grpc.Status#asException()} or
   * {@link io.grpc.Status#asRuntimeException()}. Implementations should generally convert to a
   * {@code Status} via {@link io.grpc.Status#fromThrowable(Throwable)}.
   *
   * @param t the error occurred on the stream
   */
 void onError(Throwable t);
```


From this documents we can see, `io.grpc.StatusException` or `io.grpc.StatusRuntimeException` is expected here.
I agree we need a way to handle this more gracefully.

I know there is a task planned to do better exception handling.
This PR is would serve as a temporary fix.

Another PR to do is to check the exceptions on client side, and take actions according to the exception.
For example for Status.NOT_FOUND exception, the client does not have to contact the UFS again.
This will improve performance for some use cases.

Reply:
AlluxioRuntimeException.toGrpcStatusRuntimeException would give you StatusRuntimeException with more metadata so there's no harm to use that

Reply:
Do you suggest something like this:
```
     } catch (Exception e) {
       LOG.error(String.format("Failed to list status of %s: ", request.getPath()), e);
-      responseObserver.onError(new io.grpc.StatusException(Status.UNKNOWN));
+      AlluxioRuntimeException runtimeException =
+          new AlluxioRuntimeException(Status.UNKNOWN, "Unknown", e, ErrorType.Internal, false);
+      responseObserver.onError(runtimeException.toGrpcStatusRuntimeException());
     }

```
?

Reply:
yes, or you can use `UnknownRuntimeException` which is a convenient class

Reply:
Fixed.